src/utils.ts

```diff
@@ -28,38 +29,82 @@ async function isValidWavHeader(filePath) {
 	})
 }
 
-export const convertToWavType = async (inputFilePath, logger = console) => {
-	const fileExtension = path.extname(inputFilePath).toLowerCase()
+export const convertToWavType = async (inputAudioSrc: FilePath | Buffer, logger = console) => {
+	//When Buffer data is received
+	if (Buffer.isBuffer(inputAudioSrc)) {
+		logger.debug('[Nodejs-whisper] Starting audio buffer conversion process')
 
-	logger.debug(`[Nodejs-whisper] Checking if the file is a valid WAV: ${inputFilePath}`)
+		//Create temporary input file
+		const tempInputPath = `Nodejs-whisper-input_${Date.now()}`
+		fs.writeFileSync(tempInputPath, inputAudioSrc)
+		logger.debug(`[Nodejs-whisper] Temporary input file created: ${tempInputPath}`)
```

Comment:
It looks like the output is the name of this wav file plus ".srt" or ".json" or ".vtt" etc..

So with this change having a generated filepath to the .wav would make the corresponding output file paths indeterminate.